Foundation/NSIndexSet.swift

```diff
@@ -78,7 +78,18 @@ open class NSIndexSet : NSObject, NSCopying, NSMutableCopying, NSSecureCoding {
         return copy(with: nil)
     }
     
-    open func copy(with zone: NSZone? = nil) -> Any  { NSUnimplemented() }
+    open func copy(with zone: NSZone? = nil) -> Any {
+        if type(of: self) === NSIndexSet.self {
+            // return self for immutable type
+            return self
+        } else if type(of: self) === NSMutableIndexSet.self {
+            let indexSet = NSIndexSet()
+            indexSet._ranges = self._ranges
+            indexSet._count = self._count
+            return indexSet
```

Comment:
Why wouldn't NSMutableIndexSet just override this and do the right thing, instead of having the superclass know about its subclass?